### PR TITLE
AEA-2984-Accept-both-message-formats

### PR DESCRIPTION
**Pull request by TomSmithCoding**

### CODE REVIEW

**Review by anthony-nhs — CHANGES_REQUESTED** [review]
code changes are good but could just do with some more testing.   

As well as the unit test change I have pointed out, all current examples under examples should be changed to remove practionerRole.healthCareService (as that should be the default) but there should be an example added to ensure everything works when practionerRole.healthCareService is still passed in (to ensure we do not break anything with IC24)

File: packages/coordinator/src/services/validation/bundle-validator.ts [hunk 1]
```diff
@@ -453,13 +453,8 @@ function checkPrimaryCarePrescriptionResources(
...
 
 function checkSecondaryCarePrescriptionResources(
-  practitionerRole: fhir.PractitionerRole,
   organization: fhir.Organization,
 ): fhir.OperationOutcomeIssue {
-  if (!practitionerRole.healthcareService) {
-    return errors.missingRequiredField("practitionerRole.healthcareService")
-  }
-
   if (organization.partOf) {
```
Comment (MatthewPopat-NHS) [inline packages/coordinator/src/services/validation/bundle-validator.ts:458]: this should be allowed for secondary care

File: packages/specification/schemas/MedicationRequest/extensions/NHSBSAPrescriptionType.yaml [hunk 2]
```diff
@@ -5,8 +5,6 @@ description: |
   For the full list of prescription types, and what they mean, please refer to the following [guidance](https://www.nhsbsa.nhs.uk/sites/default/files/2022-07/NHSBSAGuidanceforEndorsement_v7.7.pdf) from the BSA.
   
   Where the prescription type is of the pattern 01nn, the details of the prescribing organisation must be captured within the Organisation resource.
```
Comment (MatthewPopat-NHS) [inline packages/specification/schemas/MedicationRequest/extensions/NHSBSAPrescriptionType.yaml:7]: We should probably remove this line too

File: examples/errors/1-Process-Request-Send-400_INVALID_CHECKSUM.json [hunk 3]
```diff
@@ -358,40 +352,6 @@
         ]
       }
     },
-    {
-      "fullUrl": "urn:uuid:54b0506d-49af-4245-9d40-d7d64902055e",
```
Comment (MatthewPopat-NHS) [inline examples/errors/1-Process-Request-Send-400_INVALID_CHECKSUM.json:362]: Unfortunately for all these examples we probably shouldn't be just removing the Healthcare service resource, we should be translating it to the equivalent organisation resource.

File: packages/coordinator/src/services/validation/bundle-validator.ts (hunk 1 above)
Comment (TomSmithCoding) [reply packages/coordinator/src/services/validation/bundle-validator.ts:458]: What should be allowed sorry? We are now allowing healthcareService resources for Secondary Care prescriptions, or are we saying they should be allowed to include the `organization.partOf` to replace the healthcareService resource (I think I just answered my own question)

File: packages/specification/electronic-prescription-service-api.yaml [hunk 4]
```diff
@@ -975,7 +975,6 @@ components:
             - $ref: "#/components/schemas/Create-Practitioner-Resource"
             - $ref: "#/components/schemas/Create-MessageHeader-Resource"
             - $ref: "#/components/schemas/Create-Patient-Resource"
-            - $ref: "#/components/schemas/Create-HealthcareService-Resource"
             - $ref: "#/components/schemas/Create-Location-Resource"
```
Comment (MatthewPopat-NHS) [inline packages/specification/electronic-prescription-service-api.yaml:978]: We should remove the Location resources as well

**Review by MatthewPopat-NHS — APPROVED** [review]

**Pull request merged**